File: 11. Container With Most Water/11. Container With Most Water.c
int maxArea(int* height, int heightSize) {
    
    int cur_max_area = 0;
    int left = 0;
    int right = heightSize - 1;

    while (left < right){
        int h = height[left] < height[right] ? height[left] : height[right];
        int w = right - left;

        int area = h * w;
        if (area > cur_max_area) cur_max_area = area;

        if (height[left] < height[right]) left ++;
        else right --;
    }

    return cur_max_area;
}

File: 11. Container With Most Water/11. Container With Most Water.js
// /**
//  * @param {number[]} height
//  * @return {number}
//  */

var maxArea = function(height) {
    
    let cur_max_area = 0
    const n = height.length
    let left = 0
    let right = n - 1

    while (left < right){
        h = Math.min(height[left], height[right])
        w = right - left
        area = h * w

        if (area > cur_max_area) cur_max_area = area

        if (height[left] < height[right]) left = left + 1
        else right = right - 1
    }

    return cur_max_area
};